src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -0,0 +1,451 @@
+                    errorInvocationName,
+                    errorRecord, Environment.NewLine));
+            stringBuilder.AppendLine(errorStackTrace);
+            stringBuilder.AppendLine(string.Format("{0} {1}", PowerShellConstants.AdditionChar, errorInvocationName));
+            stringBuilder.AppendLine(string.Format("{0} {1}", PowerShellConstants.AdditionChar,
+                new string(PowerShellConstants.UnderscoreChar, invocationName.Length)));
+            stringBuilder.AppendLine(string.Format("{0}{1} {2} {3}",
+                new string(PowerShellConstants.SpaceChar, PowerShellConstants.SpaceCount),
+                PowerShellConstants.AdditionChar, PowerShellConstants.CategoryInfoLabel, errorRecord.CategoryInfo));
+            if (string.IsNullOrEmpty(invocationName))
+            {
+                stringBuilder.AppendLine(string.Format("{0}{1} {2} {3},{4}",
+                    new string(PowerShellConstants.SpaceChar, PowerShellConstants.SpaceCount),
+                    PowerShellConstants.AdditionChar, PowerShellConstants.FullyQualifiedErrorIdLabel,
+                    errorRecord.FullyQualifiedErrorId, fileName));
+            }
+            else
+            {
+                stringBuilder.AppendLine(string.Format("{0}{1} {2} {3}",
+                    new string(PowerShellConstants.SpaceChar, PowerShellConstants.SpaceCount),
+                    PowerShellConstants.AdditionChar, PowerShellConstants.FullyQualifiedErrorIdLabel,
+                    errorRecord.FullyQualifiedErrorId));
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        private string GetStackTrace(string scriptStackTrace, string fileName)
+        {
+            string stackTrace = scriptStackTrace.Replace(PowerShellConstants.StackTraceScriptBlock, fileName);
+
+            if (stackTrace.Contains(_functionName))
+            {
+                string[] tokens = stackTrace.Split((char[])null, StringSplitOptions.RemoveEmptyEntries);
+                string[] newtokens = new string[tokens.Length];
+                int index = 0;
+                foreach (string token in tokens)
+                {
+                    if (token.Contains(_functionName))
+                    {
+                        string relativePath = GetRelativePath(token);
+                        newtokens[index++] = relativePath;
+                    }
+                    else if (token.Contains(PowerShellConstants.StackTraceScriptBlock))
+                    {
+                        newtokens[index++] = token.Replace(PowerShellConstants.StackTraceScriptBlock, fileName);
+                    }
+                    else
+                    {
+                        newtokens[index++] = token;
+                    }
+                }
+
+                stackTrace = string.Join(" ", newtokens);
+            }
+
+            return stackTrace;
+        }
+
+        private string GetScript()
```

Comment:
Isn't there a way for us to just pass the path into the runspace, rather than reading this file ourselves?